enterprise/frontend/src/metabase-enterprise/advanced_permissions/components/RoleAttributeMappingModal/RoleAttributeMappingModal.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect } from "react";
+import { withRouter } from "react-router";
+import { useSelector } from "react-redux";
+import { push } from "react-router-redux";
+import { useDispatch } from "metabase/lib/redux";
+import { fetchAttributes } from "metabase-enterprise/sandboxes/actions";
+import { getAttributes } from "metabase-enterprise/sandboxes/selectors";
+import Databases from "metabase/entities/databases";
+import { State } from "metabase-types/store";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper/LoadingAndErrorWrapper";
+import { RoleAttributeMappingModalParams } from "metabase-enterprise/advanced_permissions/types";
+import { getParentPath } from "metabase/hoc/ModalRoute";
+import RoleAttributeMappingModalView from "./RoleAttributeMappingModalView";
+
+interface RoleAttributeMappingModalProps {
+  params: RoleAttributeMappingModalParams;
+  route: {
+    path: string;
+  };
+}
+
+const RoleAttributeMappingModal = ({
+  params,
+  route,
+}: RoleAttributeMappingModalProps) => {
+  const databaseId = parseInt(params.impersonatedDatabaseId);
+  const attributes = useSelector(getAttributes);
+
+  const database = useSelector((state: State) =>
+    Databases.selectors.getObject(state, { entityId: databaseId }),
+  );
```

Comment:
This isn't a comment on this PR per se, more a thought on the way to improving our entities situation: it's often hard to know when to use a selector vs. an entity loader. 

For the current usage of this component, I think we can be pretty sure we've already loaded DB data from the API, but you actually have to know a decent bit about the entire application to know that. It would be nice to have a one-size fits all way to get a database that is agnostic to whether or not the data is cached already.

Reply:
Nit: you can improve `useSelector` from `metabase/lib/redux` as well to avoid having to manually type `state`

I will try to get a PR up soon restricting imports of `useSelector` and `useDispatch` directly from `react-redux`

Reply:
Agree, actually, I think entity loaders are preferred for resilience